scripts/neighbor_advertiser

```diff
@@ -281,28 +285,53 @@ def construct_neighbor_advertiser_slice():
 
     all_vlan_interfaces = get_vlan_interfaces()
 
+    vlan_intf_table = config_db.get_table('VLAN_INTERFACE')
+
     for vlan_interface in all_vlan_interfaces:
-        vlan_id, vxlan_id, ipv4_addr, ipv6_addr, mac_addr = get_vlan_addresses(vlan_interface)
+        vlan_id, vxlan_id, ipv4_addr, ipv4_prefix, ipv6_addr, ipv6_prefix, mac_addr = get_vlan_addresses(vlan_interface)
 
         if not mac_addr:
             log.log_warning('Cannot find mac addr of vlan interface {}'.format(vlan_interface))
             continue
 
         ipv4_mappings = []
-        for addr in ipv4_addr:
-            mapping = {
-                'ipAddr': addr,
-                'macAddr': mac_addr
-            }
-            ipv4_mappings.append(mapping)
-
         ipv6_mappings = []
-        for addr in ipv6_addr:
-            mapping = {
-                'ipAddr': addr,
-                'macAddr': mac_addr
-            }
-            ipv6_mappings.append(mapping)
+        if vlan_intf_table[vlan_interface]['proxy_arp'] == 'enabled':
```

Comment:
Lets not repeat the code here, just update the prefixlen based on the 'proxy_arp' setting.